### PR TITLE
Save sparse weights in MeshTfidfSVM

### LINKED ISSUE #114: Save sparse weights in mesh tfidf-svm
`SGDClassifier` https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html which is what we use under the hood for `MeshTfidfSVM` provides a sparsity function that creates sparse weights which results in a much smaller model size. To effectively use that, it would be better to zero weights below a value say `0.001` that do not contribute much.

As a reminder the disease mesh model ended up being 70GB and if we trained on the entire mesh it would hundreds of GB. Weight pruning is a commonly used strategy to reduce model size.

Note that to use the weights for predictions you need to make them `dense` through `densify` so some care is still needed to fit into memory. Batching predictions by label as we do now will still be a good way to deal with that.

### PR DESCRIPTION
**Pull request by nsorros**
## Description

This helps us reduce size in the same that XLinear does and that I described in the blog post. Eventually MeshTfidfSVM might become an in house alternative to XLinear.

Fixes #114

## Checklist

- [x] Linked to Notion or GitHub issue
- [ ] Added tests
- [ ] Updated README
- [ ] DVC up to date

### CODE REVIEW

File: grants_tagger/models/mesh_tfidf_svm.py [hunk 1]
```diff
@@ -82,7 +89,11 @@ def fit(self, X, Y):
...
             self.classifier.fit(X_vec, Y[:, tag_i : tag_i + self.y_batch_size])
 
-            # TODO: Sparsify weights before saving
+            for i, est in enumerate(self.classifier.estimators_):
+                est.coef_[np.abs(est.coef_) < self.cutoff_prob] = 0
+                est.sparsify()
+                self.classifier.estimators_[i] = est
+
             with open(f"{self.model_path}/{tag_i}.pkl", "wb") as f:
                 f.write(pickle.dumps(self.classifier))
```
Comment (ivyleavedtoadflax) [inline grants_tagger/models/mesh_tfidf_svm.py:98]: Is there a reason for using `pickle.dumps()` rather than `pickle.dump`. If you use the latter, you don't need `f.write()`:

```
            with open(f"{self.model_path}/{tag_i}.pkl", "wb") as f:
                pickle.dump(self.classifier, f)
```

> pickle.dump(obj, file, protocol=None, *, fix_imports=True, buffer_callback=None)¶
> Write the pickled representation of the object obj to the open file object file. This is equivalent to Pickler(file, ?protocol).dump(obj).
>
> Arguments file, protocol, fix_imports and buffer_callback have the same meaning as in the Pickler constructor.
>
> Changed in version 3.8: The buffer_callback argument was added.

https://docs.python.org/3/library/pickle.html

**Review by ivyleavedtoadflax — APPROVED** [review]
Looks good. Just one comment. :shipit:

File: grants_tagger/models/mesh_tfidf_svm.py (hunk 1 above)
Comment (nsorros) [reply grants_tagger/models/mesh_tfidf_svm.py:98]: I am defaulting to this way for a while as its easier to remember for me. In other cases I was always confusing whether it was dump or dumps. Now I only use `loads` and `dumps` explicitly
